sse2neon.h

```diff
@@ -2442,7 +2499,11 @@ FORCE_INLINE __m128 _mm_setzero_ps(void)
 // https://www.intel.com/content/www/us/en/docs/intrinsics-guide/index.html#text=_mm_sfence
 FORCE_INLINE void _mm_sfence(void)
 {
+#if defined(_MSC_VER)
+    __dmb(_ARM64_BARRIER_SY);
+#else
     _sse2neon_smp_mb();
```

Comment:
Leaving a comment on this line because GitHub only lets you comment on the lines immediately surrounding changed lines...

This function is used by _mm_mfence() and _mm_lfence() below. MSVC support for stdatomic.h is still experimental and STDC_NO_ATOMICS is still defined.
https://devblogs.microsoft.com/cppblog/c11-atomics-in-visual-studio-2022-version-17-5-preview-2/

Furthermore, STDC_VERSION isn't defined at all when compiling as C++.

In both cases, _sse2neon_smp_mb() falls through to the empty "FIXME: MSVC support" else block. At a minimum, there should be a MSVC case in _sse2neon_smp_mb(). that emits the same code GCC/Clang does. Maybe these functions ought to be using more precise barriers, though.